File: backend/src/main/java/com/prabath_motors/backend/dao/Appointment/Job.java
package com.prabath_motors.backend.dao.Appointment;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "jobs")
public class Job {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String jobId;

    @Column(nullable = false)
    private String vehicleRegistrationNumber; //auto fetch from appointment

    @Column(nullable = false)
    private String serviceSection;

    @Column(nullable = false)
    private String assignedEmployee;

    @ElementCollection
    @CollectionTable(name = "job_tasks", joinColumns = @JoinColumn(name = "job_id"))
    private List<NamedCostItem> tasks; //Multiple tasks assigned to a job

    @ElementCollection
    @CollectionTable(name = "job_spare_parts", joinColumns = @JoinColumn(name = "job_id"))
    private List<NamedCostItem> spareParts;

    @Column(nullable = false)
    private String status; //Ongoing or Done


    private double totalCost; //Ongoing or Done
}

File: frontend/src/types/Job.ts
export interface Job {
    id: number;
    jobId: string;
    vehicleRegistrationNumber: string;
    serviceSection: string;
    assignedEmployee: string;
    status: string;
    tasks: NamedCostItem[];
    spareParts: NamedCostItem[];
    totalCost: number;
    ownerName: string;
    contactNo: string;
}

export interface NamedCostItem {
    name: string;
    cost: number;
}

export type JobStatus = 'Ongoing' | 'Done';

export enum ServiceSection {
    GARAGE = "GARAGE",
    BODY_SHOP = "BODY_SHOP",
    PAINT_SHOP = "PAINT_SHOP",
    WASH_BAY = "WASH_BAY"
}

export enum TaskStatus {
    PENDING = "PENDING",
    IN_PROGRESS = "IN_PROGRESS",
    COMPLETED = "COMPLETED"
}



 